Library/Homebrew/extend/kernel.rb

```diff
@@ -234,36 +234,51 @@ def odisabled(method, replacement = nil,
     odeprecated(method, replacement, disable: true, disable_on:, disable_for_developers:, caller:)
   end
 
-  sig { params(formula: T.any(String, Formula)).returns(String) }
+  sig { params(formula: Formula).returns(String) }
   def pretty_installed(formula)
+    pretty_installed_string(formula.to_s)
+  end
+
+  sig { params(string: String).returns(String) }
+  def pretty_installed_string(string)
```

Comment:
How about only supporting `pretty_installed(string)` and then convert everything passing formula to pass `pretty_installed(formula.name)`?

Reply:
Yea, `brew tc` doesn't complain about making these just accept `String`s, so maybe we never needed to pass them `Formula`e in the first place.